scheduler/scheduler-api/src/main/java/org/ow2/proactive/scheduler/common/job/factories/StaxJobFactory.java

```diff
@@ -1821,7 +1860,14 @@ private static String replace(String str, Map<String, String> variables) throws
 
         // Include given variables if any
         if (variables != null) {
-            replacements.putAll(variables);
+            for (Map.Entry<String, String> variable : variables.entrySet()) {
```

Comment:
why this code change ? could you add comments to explain ?

Reply:
Yes. If a system variable is overriden by a variable, then the system variable value should apply, correct?
This is what the following code says. Otherwise, we must decide which value preempts the other.